miden-tx/src/tests.rs

```diff
@@ -306,3 +310,151 @@ fn test_prove_and_verify_with_tx_executor() {
+                use.miden::sat::note
+                
+                begin                                                   # [note_inputs = target_account_id, ...]                                            
+                    exec.account::get_id                                # [account_id, target_account_id, ...]
+                    assert_eq                                           # [] if account_id == target_account_id, fails if not
+
+                    push.1000000000                                     # [1000000000, ...] memory pointer to store assets
+                    exec.note::get_assets                               # [num_of_assets, 1000000000, ...]   
+                    
+                    dup neq.0                                           # [1 || 0, num_of_assets, 1000000000, ...]
+                    while.true                                          # [num_of_assets, 1000000000, ...]
+                        add sub.1                                       # [num_of_assets + 1000000000 - (i), ...] -> points to last asset in memory
+                        exec.account::get_nonce drop                    # TODO: Should call add_asset but we don't have it yet
+                        dup neq.1000000000                              # [1 || 0, num_of_assets + 1000000000 - (i), ...], if end of memory reached, break
+                    end
+
+                    drop                                           # []        
+                end
+                ",
+            )
+            .as_str(),
+        )
+        .unwrap();
+    let (note_script, _) = NoteScript::new(note_program_ast, &mut assembler).unwrap();
+
+    const SERIAL_NUM: Word = [Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)];
+
+    let note = Note::new(
+        note_script.clone(),
+        &[*target_account_id],
+        &vec![fungible_asset_1],
+        SERIAL_NUM,
+        sender_account_id,
+        ONE,
+        None,
+    )
+    .unwrap();
+
+    // CONSTRUCT AND EXECUTE TX (Success)
+    // --------------------------------------------------------------------------------------------
+    let data_store = MockDataStore::new(Some(target_account), Some(vec![note.clone()]));
+    let mut executor = TransactionExecutor::new(data_store.clone());
+
+    executor.load_account(target_account_id).unwrap();
+
+    let block_ref = data_store.block_header.block_num().as_int() as u32;
+    let note_origins = data_store
+        .notes
+        .iter()
+        .map(|note| note.proof().as_ref().unwrap().origin().clone())
+        .collect::<Vec<_>>();
+
+    // execute the transaction and get the witness
+    let transaction_result =
+        executor.execute_transaction(target_account_id, block_ref, &note_origins, None);
+
+    // check that we got the expected result - TransactionResult and not TransactionExecutorError
+    match transaction_result {
+        Ok(_) => {} // expected result, we do nothing
```

Comment:
Once everything is implemented, we probably want to confirm that transaction result contains the expected data. Maybe leave a `TODO` here to revisit this later?